store/src/errors.rs

```diff
@@ -1,64 +1,63 @@
-use miden_crypto::hash::rpo::RpoDigest;
+use miden_crypto::{
+    hash::rpo::RpoDigest,
+    merkle::{MerkleError, MmrError},
+};
 use miden_node_proto::error::ParseError;
+use thiserror::Error;
+use tokio::sync::oneshot::error::RecvError;
 
-#[derive(Clone, Debug)]
+use crate::db::errors::DbError;
+
+#[derive(Error, Debug)]
 pub enum StateError {
```

Comment:
There are quite a few error variants here. I would probably try to break this up into some logical error groups. For example, we could have `ApplyBlockError` and then State::apply_block()` would return this error.

Also, a lot of variants seem to be related to the parsing of protobuf messages (e.g., `MissingAccountId`). In the future, hopefully most of them will be encapsulated in the `proto` crate (see #74), but maybe for now we just create one or two variant for errors like this (e.g., `MessageMissingField(String)`).